cmd/root.go

```diff
@@ -83,20 +84,41 @@ func Run(spec *Specification) (*event.EventEmitter, error) {
 	// Set Time from ntp
 	network.NtpDate()
 
-	err = storage.WipeDisks()
-	if err != nil {
-		return eventEmitter, errors.Wrap(err, "wipe")
-	}
-
 	reg := &register.Register{
 		MachineUUID: spec.MachineUUID,
 		Client:      client,
 		Network:     n,
 	}
 
+	hw, err := reg.ReadHardwareDetails()
+	if err != nil {
+		return eventEmitter, errors.Wrap(err, "unable to read all hardware details")
+	}
+
+	m, err := hammer.fetchMachine(spec.MachineUUID)
```

Comment:
This is always nil if the machine was never registered at metal-api before,  and registering will not happen anymore ?

Reply:
Am im wrong with my projection ?

Reply:
As with the current implementation it first reads the hardware details and returns if an error occurs. If this succeeds we fetch the machine and ignore any error, e.g. if the machine is unknown to metal-api, and proceed normally, i.e. wiping all disks, ...
In contrary, if we succeed to fetch a machine and this machine is allocated with a reinstall mark set, we skip the default workflow and proceed with the reinstallation process.